PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azuremcp/src/main/java/com/microsoft/azure/toolkit/intellij/azuremcp/AzureMcpPackageManager.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.lang3.SystemUtils;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Optional;
+
+@Slf4j
+public class AzureMcpPackageManager {
+    private final GithubClient gitHubClient;
+    private final String platform;
+
+    public AzureMcpPackageManager() {
+        this.gitHubClient = new GithubClient();
+        this.platform = getPlatformIdentifier();
+    }
+
+    @Nullable
+    public synchronized File getAzureMcpExecutable() {
+        try {
+            final GithubRelease latestRelease = gitHubClient.getLatestAzureMcpRelease();
+            if (latestRelease != null && latestRelease.getAssets() != null) {
+                final String tagName = latestRelease.getTagName();
+                log.info("Latest version of Azure MCP: " + tagName);
+
+                final Optional<GithubAsset> githubAsset = latestRelease.getAssets()
+                        .stream()
+                        .filter(asset -> asset.getName().contains(platform))
+                        .findFirst();
+
+                if (githubAsset.isPresent()) {
+                    final GithubAsset asset = githubAsset.get();
+                    log.info("Azure MCP package for current platform: " + asset.getName());
+                    final long startTime = System.currentTimeMillis();
+                    final String azMcpDir = PathManager.getPluginsPath() + "/azure-toolkit-for-intellij/azmcp";
+                    final File azMcpDirFile = new File(azMcpDir);
+                    if (azMcpDirFile.exists() || azMcpDirFile.mkdirs()) {
+                        final Path versionFile = Path.of(azMcpDir + "/version.txt");
+
+                        final File extractedDir = new File(azMcpDirFile, "/azmcp_package_" + tagName);
+                        extractedDir.mkdirs();
+                        final String executablePath = extractedDir.getAbsolutePath() + getExecutableRelativePath();
+                        final File azMcpExe = new File(executablePath);
+                        if (!azMcpExe.exists()) {
+                            Files.writeString(versionFile, tagName, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+                            final File azMcpTgz = new File(azMcpDir, "azmcp_" + tagName + ".tgz");
+                            log.info("Downloading Azure MCP Server to: " + azMcpTgz.getAbsolutePath());
+                            final boolean downloaded = gitHubClient.downloadToFile(asset.getBrowserDownloadUrl(), azMcpTgz);
```

Comment:
should we verify the signature/checksum of the downloaded file for security?

Reply:
Updated to compare the digest of the downloaded file with the digest returned by the GitHub API.